pkg/registry/core/service/storage/storage.go

```diff
@@ -250,6 +266,33 @@ func (r *GenericREST) defaultOnReadService(service *api.Service) {
 	}
 }
 
+func (r *GenericREST) afterDelete(obj runtime.Object, options *metav1.DeleteOptions) {
+	svc := obj.(*api.Service)
+
+	klog.Errorf("TIM: afterDelete: svc %v/%v - prepare to deallocate", svc.Namespace, svc.Name)
+
+	// Normally this defaulting is done automatically, but the hook (Decorator)
+	// is called at the end of this process, and we want the fully-formed
+	// object.
+	r.defaultOnReadService(svc)
+
+	// Only perform the cleanup if this is a non-dryrun deletion
+	if !dryrun.IsDryRun(options.DryRun) {
+		// It would be better if we had the caller context, but that changes
+		// this hook signature.
+		ctx := genericapirequest.WithNamespace(genericapirequest.NewContext(), svc.Namespace)
+		// TODO: can leave dangling endpoints, and potentially return incorrect
+		// endpoints if a new service is created with the same name
+		_, _, err := r.endpoints.Delete(ctx, svc.Name, rest.ValidateAllObjectFunc, &metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			klog.Errorf("delete service endpoints %s/%s failed: %v", svc.Name, svc.Namespace, err)
```

Comment:
Commit `De-layer Service REST: Delete`:
Comment is un-actionable. We should append `endpoint controller` should clean this up eventually.  

Also: Why are we deleting endpoints and not endpointSlice?

Reply:
This is a vestigial thing.  We PROBABLY can get rid of this and let the controller do it, but there was some fear that one might reuse a service name and get the old endpoints (race).  I agree we should either do both or do neither.  Will add a TODO